test/quicapitest.c

```diff
@@ -1061,6 +1061,52 @@ static int test_non_io_retry(int idx)
     return testresult;
 }
 
+static int test_no_alpn(void)
+{
+    SSL_CTX *cctx = SSL_CTX_new_ex(libctx, NULL, OSSL_QUIC_client_method());
+    SSL *clientquic = NULL;
+    QUIC_TSERVER *qtserv = NULL;
+    int testresult = 0;
+    int ret;
+
+    /*
+     * Ensure we only configure ciphersuites that are available with both the
+     * default and fips providers to get the same output in both cases
+     */
+    if (!TEST_true(SSL_CTX_set_ciphersuites(cctx, "TLS_AES_128_GCM_SHA256")))
+        goto err;
+
+    if (!TEST_ptr(cctx)
+            || !TEST_true(qtest_create_quic_objects(libctx, cctx, NULL, cert,
+                                                    privkey,
+                                                    QTEST_FLAG_FAKE_TIME,
+                                                    &qtserv,
+                                                    &clientquic, NULL)))
+        goto err;
+
+    /*
+     * Clear the ALPN we set in qtest_create_quic_objects. We use TEST_false
+     * because SSL_set_alpn_protos returns 0 for success.
+     */
+    if (!TEST_false(SSL_set_alpn_protos(clientquic, NULL, 0)))
```

Comment:
Could we use `ADD_ALL_TESTS` to run this test twice? The first time we set ALPN correctly and the second we don't. That way we know it *would* work but for the fact we don't have ALPN.